private/bufpkg/bufmodule/bufmodulestore/module_data_store.go

```diff
@@ -99,20 +105,23 @@ func ModuleDataStoreWithTar() ModuleDataStoreOption {
 /// *** PRIVATE ***
 
 type moduleDataStore struct {
-	logger *zap.Logger
-	bucket storage.ReadWriteBucket
+	logger     *zap.Logger
+	bucket     storage.ReadWriteBucket
+	filelocker filelock.Locker
 
 	tar bool
 }
 
 func newModuleDataStore(
 	logger *zap.Logger,
 	bucket storage.ReadWriteBucket,
+	filelocker filelock.Locker,
```

Comment:
I opted to go with setting a `NopLocker` when passed `nil`, since i want to keep the constructor as simple as possible and preserve the current signature. Even though there are not many callsites in the codebase, I think it's good practice and this would be representative of the behaviour we want anyway.